qiita_db/util.py

```diff
@@ -983,6 +985,36 @@ def filepath_id_to_rel_path(filepath_id):
         return result
 
 
+def filepath_id_to_object_id(filepath_id):
+    """Gets the object id to which the filepath id belongs to
+
+    Returns
+    -------
+    int
+        The object id the filepath id belongs to or None if not found
+
+    Notes
+    -----
+    This helper function is intended to be used with the download handler so
+    we can prepend downloads with the artifact id; thus, we will only look for
+    filepath ids in qiita.analysis_filepath and qiita.artifact_filepath as
+    search in qiita.reference, qiita.prep_template_filepath and
+    qiita.sample_template_filepath will make the naming redundat (those already
+    have the study_id in their filename)
+    """
+    with qdb.sql_connection.TRN:
+        sql = """
```

Comment:
what if - for strange reasons - DB tables don't exist? Are those errors handled appropriately?

Reply:
Yup, handled in the TRN object and if that happens we'll have larger and scarier issues ... :)